### ISSUE TITLE
Internationalisation [for timestamps]

### ISSUE THREAD

**Issue by UlfGran** [MythTV-Clients/MythTV-Android-Frontend#248]
I would like to be able to use a 24-hour clock format in the app. From what I have seen in the translations section it seems that currently the clock format is a part of string translations for languages. This is not really logical, and it would be better to have some kind of locale setting that makes clock and date format independent of app language. The best would of course be to use locale/time format settings from the settings in the device.

**Comment by dmfrey** [comment]
The locale settings on the device are used to determine your clock format.
This was added some time ago.

**Comment by billmeek** [comment]
Closing this issue. The existing format is ISO 8601 compliant. Please reopen if there
are additional details (e.g. screen name, specific language and format.)

**Comment by UlfGran** [comment]
Well, here are some more details: The device, a Nexus 7 with Android 4.4.2, is using Swedish as language, and Sweden as its localisation. The internal clock of the device is shown in a 24-hour format. In most other apps times are shown in 24-hour format. But in MAF version 1.20.6, all times in the Program Guide are shown in am/pm format, and hence do not follow the device settings. However, at least they are understandable, since the "am" or "pm" is clearly shown.

But in the "Upcoming Recordings" section the situation is even worse. The time and date of a recording that in the backend is set to start on january 12 at 22:30 is shown in the Upcoming Recordings section of MAF as starting on "01/12 10:30". This is ambiguous when it comes to both the time and the date, since the "am/pm" is not shown, and the date format does not follow the device setting. The stated date and time ("01/12 10:30") according to a swedish standard format would be interpreted as "december 1 10:30 am". 

There is one thing that appears to follow the device settings, though, and that is the date format in the Program Guide, i e it shows 2014-01-04 as for today.

**Comment by billmeek** [comment]
@UlfGran, thank you. I switched my "old" 4.0.4 tablet to Svenska. I always run it in the 24 hour format. All 
timestamps in Recordings, Guide and Upcoming Recordings display properly. It sounds less like a language
setting and more like a KitKat one.

Reopened.

**Comment by UlfGran** [comment]
In the "Recordings" section I get the dates according to "2014-01-05" but the time is in am/pm format, not 24-hour.
( Although it does not say "am" and "pm", it says "fm" and "em", which are the Swedish equivalents, so it is somehow a "Swedish" am/pm format. (Which in real life does not exist.)  This is the same in the Guide section, although I did not mention that above.)

I have now tried installing MAF 1.20.6 in my phone, a Sony Xperia S running Android 4.1.2, and everything I mention above is the same there. That is:
In Guide and Recordings I get dates as "2014-01-05" and times as "11:00 em/fm".
In Upcoming Recordings I get dates as "1/5" and times as "11:00".

I am not that familiar with time, date and localisation settings in Android, usually its not something I need to change. Is there something I should try changing? What and where in that case?

**Comment by billmeek** [comment]
In Settings->Data and time->Use 24-hour format mine is checked. But you've
already mentioned that your other applications properly display in that format,
so I assume that's already checked.

I did try switching Use 24-hour format off and found that I had to Force stop, Clear
cache and Clear data in the Application manager before the AM/PM were displayed.
Please do that.

**Comment by UlfGran** [comment]
Yes, the Use 24-hour format was already checked, as could be expected. And the date format was set to 2014-12-31.
I have now done the following:
I unchecked the Use 24 hour, and set the date format to 31-12-2014. I then did force-stop and clear data.
After that there was no change in neither time nor date formats shown in any of the screens in MAF.

I then checked the Use 24-hour again, and set the date format back to 2014-12-31. After force stop and clear data,
it turned out that now the 24-hour format for times was used in all MAF screens. Also, there was a noticeable change in layout of the Upcoming recordings screen. Before this was a single screen with a top to bottom list of all Upcoming recordings shown in the same screen, identical to the list that is shown in the Mythtv Android frontend status screen. But now it had changed to into a kind of single day calendar type screen where I could swipe from day to day and see the recordings for each day on a screen for each day. At the top of this screen, the date is given in 2014-12-31 format, and the times for each recording was given in 24 hour format.

 In the list of upcoming recordings that is shown on the 'MAF status screen' the times had also changed to 24 h format, but the date format remained as '1/5' as for today. It can be noted that this kind of date format is not one of the selectable options under the date and time settings.

After seeing this I repeatedly tried changing the date format to different settings and did force stop clear data between each change, but there was never any change of the date format in any of the MAF screens.

The conclusion:
- Changing the time format from and to 24h and doing a force stop/clear data between changes caused 24h format to be used in all screens.
  - The date format in all screens remains persistently unchanged by the date format setting, even if I do force stop and clear data.
  - Changing the time format and/or doing the force stop/clear data completely changed the layout of the Upcoming recordings screen.
  - The date format in the list of Upcoming recordings in the MAF status screen persistently remains in the ambiguous "1/5" type format, but in all other screens the 2014-12-31 format is persistently used.
    Regards Ulf
